chromadb/api/configuration.py

```diff
@@ -268,19 +270,70 @@ def from_legacy_params(cls, params: Dict[str, Any]) -> Self:
         for name, value in params.items():
             if name not in old_to_new:
                 raise ValueError(f"Invalid legacy HNSW parameter name: {name}")
-            new_name = old_to_new[name]
-            parameters.append(ConfigurationParameter(name=new_name, value=value))
-        return cls(parameters=parameters)
+            parameters.append(
+                ConfigurationParameter(name=old_to_new[name], value=value)
+            )
+        return cls(parameters)
+
+
+class HNSWConfigurationInterface(HNSWConfigurationInternal):
```

Comment:
nit - I find the naming "Interface" a touch confusing. Can we just comment allt he internal vs interface stuff to make this all clear?